R/pkg/inst/tests/testthat/test_sparkSQL.R

```diff
@@ -721,6 +721,16 @@ test_that(
   expect_true(dropTempView("dfView"))
 })
 
+test_that(
+  "createGlobalTempView() results in a queryable table and sql() results in a new DataFrame", {
+  df <- read.json(jsonPath)
+  createGlobalTempView(df, "global_table1")
+  newdf <- sql("SELECT * FROM global_temp.global_table1 where name = 'Michael'")
+  expect_is(newdf, "SparkDataFrame")
+  expect_equal(count(newdf), 1)
+  expect_true(dropGlobalTempView("global_table1"))
```

Comment:
I found that global temp tables are now returned by `listTables`. (R is the same).
```
scala> spark.range(10).createGlobalTempView("t1")

scala> spark.catalog.tableExists("global_temp.t1")
res8: Boolean = true

scala> spark.catalog.listTables().show
+----+--------+-----------+---------+-----------+
|name|database|description|tableType|isTemporary|
+----+--------+-----------+---------+-----------+
+----+--------+-----------+---------+-----------+
```